### PR TITLE
macOS fixes

### PR DESCRIPTION
**Pull request by nastys**
Fixes Sonic Unleashed.

### CODE REVIEW

File: .ci/build-linux.sh [hunk 1]
```diff
@@ -58,7 +58,7 @@ ninja; build_status=$?;
 
 cd ..
 
-shellcheck .ci/*.sh
+shellcheck .ci/build-linux.sh .ci/deploy-linux.sh .ci/export-azure-vars.sh .ci/export-cirrus-vars.sh .ci/github-upload.sh
```
Comment (hcorion) [inline .ci/build-linux.sh:61]: You're planning to revert this change right?

**Comment by Megamouse** [comment]
I don't see any explanation for these changes. "Fixes Sonic" isn't enough.

Anyway, The wolfssl commit should be 2b670c026debc8e70621eb5c75b60931c9aa0bc5, as it's marked as 5.1.0 stable.
Unless there's a reason to use some random commit.

**Comment by illusion0001** [comment]
You can force push to restart ci

**Comment by nastys** [comment]
> You can force push to restart ci

? That's not why I closed the PR.

**Comment by illusion0001** [comment]
What are the reasons then 😛

File: rpcs3/rpcs3qt/gui_application.cpp [hunk 2]
```diff
@@ -105,6 +106,20 @@ bool gui_application::Init()
 		welcome->exec();
 	}
 
+#ifndef _WIN32
+	//Check maxfiles
+	if(utils::get_maxfiles() < 4096)
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/gui_application.cpp:111]:
```suggestion
	// Check maxfiles
	if (utils::get_maxfiles() < 4096)
```

File: rpcs3/rpcs3qt/gui_application.cpp [hunk 3]
```diff
@@ -332,11 +347,13 @@ void gui_application::InitializeCallbacks()
 			g_fxo->init<rsx::thread, named_thread<NullGSRender>>();
 			break;
 		}
+#if not defined(__APPLE__)
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/gui_application.cpp:350]: I don't see how this single ifdef around the opengl renderer makes any sense

File: rpcs3/main.cpp [hunk 4]
```diff
@@ -508,6 +508,19 @@ int main(int argc, char** argv)
 	setenv( "KDE_DEBUG", "1", 0 );
 #endif
 
+#ifdef __APPLE__
+	struct ::rlimit rlim;
+	::getrlimit(RLIMIT_NOFILE, &rlim);
+	rlim.rlim_cur = OPEN_MAX;
+	if (::setrlimit(RLIMIT_NOFILE, &rlim) != 0)
+		std::cerr << "Failed to set max open file limit (" << OPEN_MAX << ").\n";
```
Comment (Megamouse) [inline rpcs3/main.cpp:516]: why is this cerr

File: rpcs3/util/sysinfo.hpp [hunk 5]
```diff
@@ -3,6 +3,16 @@
...
 
+#ifdef __APPLE__
+// sysinfo_darwin.mm
+namespace Darwin_Version
+{
+	extern int getNSmajorVersion();
+	extern int getNSminorVersion();
+	extern int getNSpatchVersion();
+}
+#endif
```
Comment (Megamouse) [inline rpcs3/util/sysinfo.hpp:14]: why is this section here if it's only used in the cpp anyway?

File: rpcs3/main.cpp (hunk 4 above)
Comment (nastys) [reply rpcs3/main.cpp:516]: I copied that from the Linux code.

File: rpcs3/rpcs3qt/gui_application.cpp [hunk 6]
```diff
@@ -105,6 +106,20 @@ bool gui_application::Init()
...
+	// Check maxfiles
+	if (utils::get_maxfiles() < 4096)
+	{
+		QMessageBox::warning(nullptr,
+							 tr("Warning"),
+							 tr("The current limit of maximum file descriptors is too low.\n"
+								"Some games will crash.\n"
+								"\n"
+								"Please increase the limit before running RPCS3."),
+							 QMessageBox::Ok, QMessageBox::NoButton);
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/gui_application.cpp:119]: These two arguments are the default and can be omitted.

File: rpcs3/rpcs3qt/render_creator.cpp [hunk 7]
```diff
@@ -104,6 +104,19 @@ render_creator::render_creator(QObject *parent) : QObject(parent)
...
+#ifdef __APPLE__
+	OpenGL.supported = false;
+
+	if (!Vulkan.supported)
+	{
+		QMessageBox::warning(nullptr,
+							 tr("Warning"),
+							 tr("Vulkan is not supported on this Mac.\n"
+								"No graphics will be rendered."),
+							 QMessageBox::Ok, QMessageBox::NoButton);
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/render_creator.cpp:116]: These two arguments are the default and can be omitted.

File: rpcs3/main.cpp (hunk 4 above)
Comment (Jayark365) [reply rpcs3/main.cpp:516]: This seems to have fixed a couple games with the Open File limit error. Notably `Tony Hawk's® Pro Skater™ 5 [BLUS31568]` Keep up the great work Nastys!!!!

Comment (JohnHolmesII) [reply rpcs3/main.cpp:516]: It was likely done like that for linux since the app will be opened via the commandline, so stderr is highly visible. Not sure that makes sense on macos.

Comment (nastys) [reply rpcs3/main.cpp:516]: You can open it on macOS in a terminal window, and you don't necessarily have to open it in a terminal window on Linux.

**Comment by Jayark365** [comment]
Would it possible for you to add support for the 8bitdo controllers both usb and bluetooth?

File: rpcs3/main.cpp (hunk 4 above)
Comment (Megamouse) [reply rpcs3/main.cpp:516]: The idea is that we can see this in the log, instead of just relying on insufficient user statements.
You can do both cerr and log error for both occasions.

Comment (nastys) [reply rpcs3/main.cpp:516]: Line 521 should log maxfiles to file on both macOS and Linux.

**Comment by DTibor1986** [comment]
> 

Xbox Series controller support would be also appreciated.

**Comment by mavethee** [comment]
@Jayark365 @DTibor1986 This is more like a Feature Request to do, controller support in general isn't a part of this PR because those controllers aren't supported in RPCS3 as a whole.

**Comment by Megamouse** [comment]
I think this was all merged already, so I'll close it to prevent offtopic comments

**Comment by Jayark365** [comment]
> I think this was all merged already, so I'll close it to prevent offtopic comments

merged into where exactly? according to GitHub, it was never merged anywhere, just closed.

**Comment by MSuih** [comment]
Search for closed PRs from the same author and you'll find that this huge changeset was split into list of small pull requests that have all been merged now.
